delocate/tools.py

```diff
@@ -246,18 +369,30 @@ def get_install_names(filename):
     install_names : tuple
         tuple of install names for library `filename`
     """
-    lines = _cmd_out_err(["otool", "-L", filename])
-    if not _line0_says_object(lines[0], filename):
+    otool = subprocess.run(
+        ["otool", "-L", filename],
+        stdout=subprocess.PIPE,
+        stderr=subprocess.PIPE,
+        universal_newlines=True,
+    )
+    if not _line0_says_object(otool.stdout or otool.stderr, filename):
         return ()
-    names = tuple(parse_install_name(line)[0] for line in lines[1:])
-    install_id = get_install_id(filename)
-    if install_id is not None:
-        assert names[0] == install_id
-        return names[1:]
-    return names
-
-
-def get_install_id(filename):
+    install_ids = _get_install_ids(filename)
+    all_names = []
```

Comment:
Comment here to say that we're collecting all libraries we depend on, for any architecture.

(Is that always the right thing to do)?

Reply:
I'm not sure if this is correct.  I could do what I did in the other functions and throw errors if different architectures have different results.

Reply:
I've made these functions more strict.